adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicRDD.scala

```diff
@@ -539,6 +654,45 @@ trait GenomicRDD[T, U <: GenomicRDD[T, U]] {
   }
 
   /**
+   * All shuffleRegionJoins share the same code to prepare, so this is a helper
```

Comment:
Stylistic nit: doc comments should start with a short (1 sentence, preferably <80 chars) description of the method, then a paragraph break, then the detailed docs.